src/ir/ty.rs

```diff
@@ -568,7 +568,8 @@ impl Type {
         let canonical_ty = ty.canonical_type();
         let kind = match ty.kind() {
             CXType_Unexposed if *ty != canonical_ty &&
-                                canonical_ty.kind() != CXType_Invalid => {
+                                canonical_ty.kind() != CXType_Invalid &&
+                                !canonical_ty.spelling().contains("type-parameter") => {
```

Comment:
Even better for future readability if we can wrap this whole condition up in a `type_is_not_canonical` helper function or something along those lines.